sklearn/linear_model/coordinate_descent.py

```diff
@@ -2033,3 +2034,138 @@ def __init__(self, eps=1e-3, n_alphas=100, alphas=None, fit_intercept=True,
             max_iter=max_iter, tol=tol, copy_X=copy_X,
             cv=cv, verbose=verbose, n_jobs=n_jobs, random_state=random_state,
             selection=selection)
+
+class AdaptiveLasso(Lasso):
+    """
+    The multi-step adaptive Lasso iteratively solves Lasso estimates with 
+    penalty weights applied to the regularization of the coefficients.
+    The optimization objective for the adaptive Lasso is::
+        1/n * ||y - X Beta||^2_2 + alpha * w |Beta|_1
+
+    Where w is a weight vector calculated in the previous stage by::
+        w_j = alpha/(|Beta_j|^gamma + eps)
```

Comment:
you should clarify what cost function you're actually minimizing with this reweighting scheme.

see this old snippet of mine:

https://gist.github.com/agramfort/1610922